src/common/templates/_initContainer.tpl

```diff
@@ -61,4 +61,35 @@ Params:
     - "pod"
     - "-lapp={{ .appName }}"
 {{- end }}
+{{- end }}
+{{/*
+Create an initContainer to copy files from a source to a destination.
+
+Usage:
+{{ include "harnesscommon.initContainer.setupWritable" (dict "image" .Values.image "imagePullPolicy" .Values.image.pullPolicy "sourcePath" "/opt/harness" "destinationPath" "/shared/volume" "volumeName" "harness-opt" "securityContext" .Values.securityContext "context" $) }}
+
+Params:
+  - image: String - Required. Container image.
+  - imagePullPolicy: String - Optional. Image pull policy.
+  - sourcePath: String - Required. Source path to copy from.
+  - destinationPath: String - Required. Destination path to copy to.
+  - volumeName: String - Required. Name of the volume to mount.
+  - securityContext: Object - Optional. Security context for the container.
+*/}}
+{{- define "harnesscommon.initContainer.setupWritable" -}}
+{{- $values := .root.Values }}
+- name: setup-harness-writable
+  image: {{ include "common.images.image" (dict "imageRoot" $values.image "global" .root.global) }}
```

Comment:
This would mean it will always use the root container image, which cannot be overridden. Thus, the scope of this function is pretty much limited to copying files from the core image container. It's not a problem, but it's just to keep in mind.